### PR TITLE
1824 claim history

### PR DESCRIPTION
**Pull request by ckr123**

### CODE REVIEW

File: domains/transfer/Transfer.API/API/Program.cs [hunk 1]
```diff
@@ -86,11 +92,67 @@
...
+                    case "Update":
+                    {
+                        historyEntity.TransferAgreementId = (Guid)eventEntry.PrimaryKey.Values.First();
+                        break;
+                    }
+                }
+
+                return true;
+            })
+            .Map<ClaimSubject, ClaimSubjectHistory>((evt, eventEntry, historyEntity) =>
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API/Program.cs:136]: This is being done twice? Here and in startup. I would expect it only to be in startup

File: domains/transfer/Transfer.API/API/Transfer/Startup.cs [hunk 2]
```diff
@@ -22,53 +23,26 @@ public static class Startup
...
-        services.AddOptions<OtlpOptions>().BindConfiguration(OtlpOptions.Prefix).ValidateDataAnnotations().ValidateOnStart();
+        services.AddOptions<ProjectOriginOptions>().BindConfiguration(ProjectOriginOptions.ProjectOrigin)
+            .ValidateDataAnnotations().ValidateOnStart();
+        services.AddOptions<OtlpOptions>().BindConfiguration(OtlpOptions.Prefix).ValidateDataAnnotations()
+            .ValidateOnStart();
 
-        Configuration.Setup()
-            .UseEntityFramework(ef => ef
-                .AuditTypeExplicitMapper(config => config
-                    .Map<TransferAgreement, TransferAgreementHistoryEntry>((evt, eventEntry, historyEntity) =>
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API/Transfer/Startup.cs:31]: Why has this been moved to program.cs?

File: domains/transfer/Transfer.API/API/Claiming/Api/Repositories/IClaimRepository.cs [hunk 3]
```diff
@@ -11,4 +11,5 @@ public interface IClaimRepository
 
     Task<ClaimSubject> AddClaimSubject(ClaimSubject claimSubject);
     void DeleteClaimSubject(ClaimSubject claim);
+    Task<List<ClaimSubjectHistory>> GetHistory(string subject);
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API/Claiming/Api/Repositories/IClaimRepository.cs:14]: Guid subject

File: domains/transfer/Transfer.API/API.IntegrationTests/Claiming/Controllers/ClaimControllerTest.cs [hunk 4]
```diff
@@ -70,6 +74,57 @@ await factory.SeedClaims(new List<ClaimSubject>()
         result.StatusCode.Should().Be(HttpStatusCode.OK);
     }
 
+    [Fact]
+    public async Task GetClaimSubjectHistory_WhenHistoryNotExist_ReturnsNotFound()
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API.IntegrationTests/Claiming/Controllers/ClaimControllerTest.cs:78]: _WhenHistoryEntryHasDifferentSubjectId

File: domains/transfer/Transfer.API/API.IntegrationTests/Claiming/Controllers/ClaimControllerTest.cs [hunk 5]
```diff
@@ -70,6 +74,57 @@ await factory.SeedClaims(new List<ClaimSubject>()
...
+    {
+
+        var subject = Guid.NewGuid();
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API.IntegrationTests/Claiming/Controllers/ClaimControllerTest.cs:81]: remove this for readability  and use Guid.NewGuid() directly

File: domains/transfer/Transfer.API/API/Transfer/Startup.cs (hunk 2 above)
Comment (ckr123) [reply domains/transfer/Transfer.API/API/Transfer/Startup.cs:31]: It seems that Audit.net only can have one Configuration.Setup, but i might be able to move to mapper to the Startup - i will look into it.

File: domains/transfer/Transfer.API/API/Program.cs (hunk 1 above)
Comment (ckr123) [reply domains/transfer/Transfer.API/API/Program.cs:136]: Moved to only program

File: domains/transfer/Transfer.API/API/Transfer/Startup.cs (hunk 2 above)
Comment (ckr123) [reply domains/transfer/Transfer.API/API/Transfer/Startup.cs:31]: Audit.NET is now used in different sub systems

**Review by tnickelsen — APPROVED** [review]

**Pull request merged**